ArduPlane/fence.cpp

```diff
@@ -193,6 +203,11 @@ bool Plane::fence_stickmixing(void) const
 
 bool Plane::in_fence_recovery() const
 {
+    if (control_mode == &mode_auto && !mission.get_in_landing_sequence_flag()) {
```

Comment:
I was surprised to see that on master we have a separate boolean for `in_return_path()` - I think we might need a clause in here for that state.

Reply:
it is tricky to see how that will work, the flag gets set even if we're not using RET_AUTOLAND=4
we'd need to check for the flag not being set

Reply:
Perhaps we should say the current patches are a significant improvement and create an issue for this case, then?

Reply:
yes, I think so